tests/e2e/testsuites/dynamically_provisioned_resize_volume_tester.go

```diff
@@ -0,0 +1,92 @@
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"fmt"
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/util"
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/tests/e2e/driver"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+// DynamicallyProvisionedResizeVolumeTest will provision required StorageClass(es), PVC(s) and Pod(s)
+// Waiting for the PV provisioner to create a new PV
+// Update pvc storage size
+// Waiting for new PVC and PV to be ready
+// And finally attach pvc to the pod and wait for pod to be ready.
+type DynamicallyProvisionedResizeVolumeTest struct {
+	CSIDriver driver.DynamicPVTestDriver
+	Pod       PodDetails
+}
+
+func (t *DynamicallyProvisionedResizeVolumeTest) Run(client clientset.Interface, namespace *v1.Namespace) {
+	volume := t.Pod.Volumes[0]
+	tpvc, _ := volume.SetupDynamicPersistentVolumeClaim(client, namespace, t.CSIDriver)
+	defer tpvc.Cleanup()
+
+	pvcName := tpvc.persistentVolumeClaim.Name
+	pvc, err := client.CoreV1().PersistentVolumeClaims(namespace.Name).Get(pvcName, metav1.GetOptions{})
+	By(fmt.Sprintf("Get pvc name: %v", pvc.Name))
+	originalSize := pvc.Spec.Resources.Requests["storage"]
+	delta := resource.Quantity{}
+	delta.Set(util.GiBToBytes(1))
+	originalSize.Add(delta)
+	pvc.Spec.Resources.Requests["storage"] = originalSize
+
+	By("resizing the pvc")
+	updatedPvc, err := client.CoreV1().PersistentVolumeClaims(namespace.Name).Update(pvc)
+	if err != nil {
+		framework.ExpectNoError(err, fmt.Sprintf("fail to resize pvc(%s): %v", pvcName, err))
+	}
+	updatedSize := updatedPvc.Spec.Resources.Requests["storage"]
+
+	By("checking the resizing PV result")
+	error := WaitForPvToResize(client, namespace, updatedPvc.Spec.VolumeName, updatedSize, 1*time.Minute, 5*time.Second)
+	framework.ExpectNoError(error)
+
+	By("Validate volume can be attached")
+	tpod := NewTestPod(client, namespace, t.Pod.Cmd)
+
+	tpod.SetupVolume(tpvc.persistentVolumeClaim, volume.VolumeMount.NameGenerate+"1", volume.VolumeMount.MountPathGenerate+"1", volume.VolumeMount.ReadOnly)
+
+	By("deploying the pod")
+	tpod.Create()
+	By("checking that the pods is running")
+	tpod.WaitForSuccess()
+
+	defer tpod.Cleanup()
+
+}
+
+// WaitForPvToResize waiting for pvc size to be resized to desired size
+func WaitForPvToResize(c clientset.Interface, ns *v1.Namespace, pvName string, desiredSize resource.Quantity, timeout time.Duration, interval time.Duration) error {
+	By(fmt.Sprintf("Waiting up to %v for pv in namespace %q to be complete", timeout, ns.Name))
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(interval) {
+		newPv, _ := c.CoreV1().PersistentVolumes().Get(pvName, metav1.GetOptions{})
+		newPvSize := newPv.Spec.Capacity["storage"]
+		if desiredSize.Equal(newPvSize) {
```

Comment:
you also return an error if they're not equal, right?

Reply:
Yeah, it will return error if they are not equal and timeout at line 91

Reply:
Ah I see we can't exit early.